plugin/kotel/meter_test.go

```diff
@@ -0,0 +1,39 @@
+package kotel
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+)
+
+func TestWithMeter(t *testing.T) {
+	provider := metric.NewNoopMeterProvider()
+
+	testCases := []struct {
+		name string
+		opts []MeterOpt
+		want *Meter
+	}{
+		{
+			name: "With MeterProvider",
+			opts: []MeterOpt{MeterProvider(provider)},
+			want: &Meter{
+				provider: provider,
+				meter: provider.Meter(
+					instrumentationName,
+					metric.WithInstrumentationVersion(semVersion()),
```

Comment:
... sorry for the back and forth here but, does `SemVersion` actually need to be public to end users? I don't know myself

Reply:
End-users may not need `SemVersion` as it's mainly used for internal tracking and tagging of the traces/metrics. Do you have a use case in mind?

Semi-related: end-users will create providers to create meters and tracers. The same providers would most likely be passed in kotel, as we demonstrated in the example. Should these meters and tracers be exposed so things outside kotel are included in the "Instrumentation Library"? This decision leads to other things such as providing predefined process tracers with defaults as asked by end-users https://github.com/twmb/franz-go/issues/213#issuecomment-1269267559.

Currently, only the meters and tracers created in kotel and started in the hooks are considered part of the library. Please take a look at the trace export to understand what I'm saying.

```
{
        "Name": "topic send",
        ...
        "InstrumentationLibrary": {
                "Name": "github.com/twmb/franz-go/plugin/kotel",
                "Version": "semver:v0.0.0-00010101000000-000000000000",
                "SchemaURL": "https://opentelemetry.io/schemas/1.12.0"
        }
}
{
        "Name": "request",
        ...
        "InstrumentationLibrary": {
                "Name": "request-service",
                "Version": "",
                "SchemaURL": ""
        }
}
{
        "Name": "topic receive",
        ...
        "InstrumentationLibrary": {
                "Name": "github.com/twmb/franz-go/plugin/kotel",
                "Version": "semver:v0.0.0-00010101000000-000000000000",
                "SchemaURL": "https://opentelemetry.io/schemas/1.12.0"
        }
}
{
        "Name": "topic process",
        ...
        "InstrumentationLibrary": {
                "Name": "process-service",
                "Version": "",
                "SchemaURL": ""
        }
}
```

Reply:
> Do you have a use case in mind?

No use case. If it's not obviously needed by end users, then I err on keeping it private for now until there's a need.

> Should these meters and tracers be exposed so things outside kotel are included in the "Instrumentation Library"?

What would the API look like? Not following from the example but I think an API would be a bit clearer.

Reply:
Based on my understanding, the `semVersion` is currently set to private.


### 1) Idea 1 - Clients use kotel.Tracer

To expose the Kotel tracer, we can make the Tracer field public. This will allow users to create spans using the Kotel tracer like so:

```
kotelTracer := kotel.NewTracer([]kotel.TracerOpt{}...)
ctx, span := kotelTracer.Tracer.Start(context.Background(),"awsomeSpan", []trace.SpanStartOption{}...)
```

### 2) Idea 2 - Add WithProcessTracer Method to kotel.Tracer

We can also add a new method `WithProcessTracer` to the Tracer struct, to provide users with a tracer that is up to the Otel messaging specification.

```
func (t *Tracer) WithProcessTracer(r *kgo.Record) (context.Context, trace.Span) {
	// Set up the span options.
	attrs := []attribute.KeyValue{
		semconv.MessagingSystemKey.String("kafka"),
		semconv.MessagingDestinationKindTopic,
		semconv.MessagingDestinationKey.String(r.Topic),
		semconv.MessagingOperationProcess,
		semconv.MessagingKafkaPartitionKey.Int64(int64(r.Partition)),
	}
	if r.Key != nil {
		attrs = append(attrs, semconv.MessagingKafkaMessageKeyKey.String(string(r.Key)))
	}
	if t.clientID != "" {
		attrs = append(attrs, semconv.MessagingKafkaClientIDKey.String(t.clientID))
	}
	if t.consumerGroup != "" {
		attrs = append(attrs, semconv.MessagingKafkaConsumerGroupKey.String(t.consumerGroup))
	}
	opts := []trace.SpanStartOption{
		trace.WithAttributes(attrs...),
		trace.WithSpanKind(trace.SpanKindConsumer),
	}

	if r.Context == nil {
		r.Context = context.Background()
	}
	// Start a new span using the provided context and options.
	return t.tracer.Start(r.Context, r.Topic+" process", opts...)
}
```

It would be called like so:

```
func processRecord(record *kgo.Record, tracer kotel.Tracer) {
	_, span := tracer.ProcessTracer(record)
	// Simulate some work.
	time.Sleep(1 * time.Second)
	// End the span when function exits.
	defer span.End()
	fmt.Printf(
		"processed offset '%s' with key '%s' and value '%s'\n",
		strconv.FormatInt(record.Offset, 10),
		string(record.Key),
		string(record.Value),
	)
}
```

### Conclusion:
In conclusion, for idea 2, If we want to encapsulate all 3 spans required from the Otel messaging spec, then the simplest way I can think of is creating a WithProcessTracer method. It will save our users a step from creating this in their own code and guarantees all the attributes are set correctly. However, for idea 1, we should not do this and leave it as is. It makes sense to separate the boundary between internal and outside instrumentation. Users can continue to create their tracers, and their applications using their instrumentation library.

Let me know what you think about idea 2. I can make this change in the PR.

Reply:
I like the second idea more -- if I'm reading it correctly, it allows us to do things the right way and hide away complexity from the user. Is there a reason people might want the control of the first option?

Reply:
No, they can create their own tracers. :)